multicluster/controllers/multicluster/commonarea/remote_common_area_test.go

```diff
@@ -84,4 +154,65 @@ func TestMemberAnnounce(t *testing.T) {
 	}
 
 	remoteCommonAreaUnderTest.Stop()
+	status := remoteCommonAreaUnderTest.GetStatus()
+
+	assert.Equal(t, status[0].Status+status[1].Status, v1.ConditionTrue+v1.ConditionTrue)
+	assert.Equal(t, status[0].Reason+status[1].Reason, "")
+	assert.Equal(t, status[0].Message+status[1].Message, "This leader cluster is the leader for local cluster")
+	assert.Equal(t, status[0].Type+status[1].Type, mcsv1alpha1.ClusterReady)
+}
+
+func TestMemberAnnounceNewRemoteCommonArea(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	mockManager := mocks.NewMockManager(mockCtrl)
+	fakeRemoteClient := fake.NewClientBuilder().WithScheme(scheme).Build()
+
+	remoteCommonExpect := &remoteCommonArea{
+		Client:             fakeRemoteClient,
+		ClusterManager:     mockManager,
+		ClusterSetID:       "clusterSetA",
+		ClusterID:          "leaderA",
+		config:             nil,
+		scheme:             scheme,
+		Namespace:          "cluster-a-ns",
+		connected:          false,
+		localClusterClient: nil,
+		localNamespace:     "localnamespace",
+		localClusterID:     "clusterA",
+		clusterStatus: mcsv1alpha1.ClusterCondition{
+			Type:    mcsv1alpha1.ClusterReady,
+			Status:  v1.ConditionUnknown,
+			Message: "Leader cluster added",
+		},
+		leaderStatus: mcsv1alpha1.ClusterCondition{
+			Type:    mcsv1alpha1.ClusterIsLeader,
+			Status:  v1.ConditionFalse,
+			Message: "Leader cluster added",
+		},
+	}
+
+	remote, err := NewRemoteCommonArea(remoteCommonExpect.ClusterID, remoteCommonExpect.ClusterSetID, remoteCommonExpect.localClusterID, mockManager, fakeRemoteClient, scheme, nil,
+		"cluster-a-ns", "localnamespace", nil)
+	assert.Equal(t, err, nil)
+	clusterStatus, leaderStatus := remote.GetStatus()[0], remote.GetStatus()[1]
+	remoteCommonExpect.clusterStatus.LastTransitionTime = clusterStatus.LastTransitionTime
+
+	remoteCommonExpect.leaderStatus.LastTransitionTime = leaderStatus.LastTransitionTime
+	assert.Equal(t, remote, remoteCommonExpect)
+}
+
+func TestMemberAnnounceGetSecretCACrtAndToken(t *testing.T) {
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "mcs1",
+			Name:      "member-token",
+		},
+		Data: map[string][]byte{
+			"ca.crt": []byte(`12345`),
+			"token":  []byte(`12345`)},
+	}
+	caData, token, err := GetSecretCACrtAndToken(secret)
+	assert.Equal(t, caData, []byte(`12345`))
+	assert.Equal(t, token, []byte(`12345`))
+	assert.Equal(t, err, nil)
```

Comment:
The first one is the expected result. Please check other occurrences.
```suggestion
	assert.Equal(t, nil, err)
```